packages/core-components/src/hooks/useQueryParamState.ts

```diff
@@ -75,7 +75,7 @@ export function useQueryParamState<T>(
     );
   }, [searchParamsString, setQueryParamState, stateName]);
 
-  useDebounce(
+  useDebouncedCallback(
```

Comment:
I believe this should be `useDebouncedEffect` instead? Right now I don't think this code ends up doing much as the returned callback isn't called

Reply:
Ive changed it now. The hook `useQueryParamState` seems to be only used by a deprecated LegacySearchPage, although it is also exported so perhaps its still used outside of the LegacySearchPage.